Iguana2024/devops-2024
# Igd 7 set up ssh key authentication and security audit

## What type of PR is this? (check all applicable)


## Description
Added ssh-key authentication logic to Vagrantfile which includes adding users depending on public-keys(in format username.pub), as well as adding logic to take your private-key for communication with other vm's.
Also fixed the line 6 " sudo  apt-get install -y vsftpd rkhunter" in "provision.sh" with "DEBIAN_FRONTEND=noninteractive"
Ensured that security audit is starting at the start of the vm by adding 
"sudo rkhunter --update
sudo rkhunter --propupd
sudo rkhunter --check --checkall"
on lines 36,37 and 38 in "provision.sh"

## QA Instructions, Screenshots, Recordings
Check changes by:
1.ssh -i your_private_key your_username@192.168.56.101
![image](https://github.com/Iguana2024/devops-2024/assets/64985383/2bcd2543-f1f3-4aa8-ac06-fe44f496043d)
2.Afther that you should login as the user
![image](https://github.com/Iguana2024/devops-2024/assets/64985383/2f89450c-a5ba-438a-8250-408bd90686d5)
3.To check communication between machines:
login as user vagrant
![image](https://github.com/Iguana2024/devops-2024/assets/64985383/061dcd78-7531-48e1-8611-a74b82617673)
and type in ssh vagrant@ip_of_other_machine
![image](https://github.com/Iguana2024/devops-2024/assets/64985383/38dc3579-03e8-4448-a721-2bcb5a1a196b)
if you are on the other machine than everything is working
![image](https://github.com/Iguana2024/devops-2024/assets/64985383/eaf4a774-a6ef-49ac-9b2b-e70f36067622)



## [optional] Are there any post deployment tasks we need to perform?
